lib/sky-pages-assets-generator.js

```diff
@@ -29,11 +32,40 @@ function getSource() {
 
     return pathMap[filePath];
   }
+
+  public getResourcesForLocale(locale: string): any {
+    const resources: {[key: string]: any} = ${localesSource};
+    return resources[locale];
+  }
 }`;
 
   return src;
 }
 
+function getLocalesSource() {
+  const localesPath = ['src', 'assets', 'locales', 'resources_*.json'];
+  const files = glob.sync(skyPagesConfigUtil.spaPath(...localesPath))
+    .concat(
+      glob.sync(skyPagesConfigUtil.spaPath(
+        'node_modules', '@blackbaud', '**', ...localesPath
+      ))
+    );
+
+  const resources = {};
+  files.forEach((filePath) => {
+    const key = filePath.split('resources_')[1].split('.json')[0];
+    const contents = fs.readJSONSync(filePath);
+
+    if (resources[key]) {
+      resources[key] = Object.assign(resources[key], contents);
```

Comment:
I'm confused by this line.  Why assign it to the target and return it?

Reply:
This line merges resources that share the same locale (in this case, if the library and the spa both have an `en_US` file, it will store the strings under the same `en_US` key).